x-pack/plugins/enterprise_search/server/lib/check_access.ts

```diff
@@ -44,7 +44,7 @@ export const checkAccess = async ({
   request,
   log,
 }: CheckAccess): Promise<ProductAccess> => {
-  const isRbacEnabled = security.authz.mode.useRbacForRequest(request);
+  const isRbacEnabled = security?.authz.mode.useRbacForRequest(request);
```

Comment:
should we just add a:

```typescript
if (security === undefined) {
  return DENY_ALL_PLUGINS;
}
```

to simplify the code a bit.

Reply:
Went with Oleg's suggestion that the security plugin is always available instead, which obviates this.